packages/@n8n/ai-workflow-builder.ee/src/tools/get-node-parameter.tool.ts

```diff
@@ -0,0 +1,152 @@
+import { tool } from '@langchain/core/tools';
+import type { Logger } from '@n8n/backend-common';
+import get from 'lodash/get';
+import type { INode, NodeParameterValueType } from 'n8n-workflow';
+import { z } from 'zod';
+
+import { MAX_PARAMETER_VALUE_LENGTH } from '@/constants';
+import {
+	createNodeParameterTooLargeError,
+	getCurrentWorkflow,
+	getWorkflowState,
+} from '@/tools/helpers';
+
+import { ValidationError, ToolExecutionError } from '../errors';
+import { createProgressReporter, reportProgress } from './helpers/progress';
+import { createSuccessResponse, createErrorResponse } from './helpers/response';
+import { validateNodeExists, createNodeNotFoundError } from './helpers/validation';
+import type { GetNodeParameterOutput } from '../types/tools';
+
+/**
+ * Schema for getting specific node parameter
+ */
+const getNodeParameterSchema = z.object({
+	nodeId: z.string().describe('The ID of the node to extract parameter value'),
+	path: z
+		.string()
+		.describe('Path to the specific parameter to extract, e.g., "url" or "options.baseUrl"'),
+});
+
+function extractParameterValue(node: INode, path: string): NodeParameterValueType | undefined {
+	const nodeParameters = node.parameters;
+
+	return get(nodeParameters, path);
+}
```

Comment:
how should we handle default values when the value is undefined?
If this is an expression, should it return the expression or the resolved value?

Reply:
This tool works with parameters in a sense of "what's in the workflow JSON", so it doesn't handle default values, etc. It was added as a backup way for agent to "see" full values of the parameters omitted by workflow JSON trimming.